c++/src/kj/async-io-unix.c++

```diff
@@ -2018,24 +2008,15 @@ public:
 
   PipeThread newPipeThread(
       Function<void(AsyncIoProvider&, AsyncIoStream&, WaitScope&)> startFunc) override {
-    int fds[2]{};
-    int type = SOCK_STREAM;
-#if __linux__ && !__BIONIC__
-    type |= SOCK_NONBLOCK | SOCK_CLOEXEC;
-#endif
-    KJ_SYSCALL(socketpair(AF_UNIX, type, 0, fds));
-
-    int threadFd = fds[1];
-    KJ_ON_SCOPE_FAILURE(close(threadFd));
-
-    auto pipe = lowLevel.wrapSocketFd(fds[0], NEW_FD_FLAGS);
+    auto socketpair = newOsSocketpair();
 
-    auto thread = heap<Thread>([threadFd,startFunc=kj::mv(startFunc)]() mutable {
+    auto pipe = lowLevel.wrapSocketFd(kj::mv(socketpair.fds[0]), NEW_FD_FLAGS);
+    auto thread = heap<Thread>([threadFd=kj::mv(socketpair.fds[1]),startFunc=kj::mv(startFunc)]() mutable {
       UnixEventPort eventPort;
       EventLoop eventLoop(eventPort);
       WaitScope waitScope(eventLoop);
       LowLevelAsyncIoProviderImpl lowLevel(eventPort);
```

Comment:
The original commit uses `LowLevelAsyncIoProviderImpl lowLevelImpl; LowLevelAsyncIoProvider& lowLevel = lowLevelImpl;` here, I did not understand what that was for so I omitted it for now.